addons/stock/models/stock_quant.py

```diff
@@ -178,8 +178,9 @@ def _update_dict(date_by_quant, key, value):
 
     def _search(self, domain, *args, **kwargs):
         domain = [
-            line if not isinstance(line, (list, tuple)) or not line[0].startswith('lot_properties.')
-            else ['lot_id', 'any', [line]]
+            ['lot_id', 'any', [line]]
+            if isinstance(line, (list, tuple)) and isinstance(line[0], str) and line[0].startswith('lot_properties.')
```

Comment:
Well... Since the idea is to make it more robust, let's do it as much as possible
```suggestion
            if line and isinstance(line, (list, tuple)) and isinstance(line[0], str) and line[0].startswith('lot_properties.')
```
(In case `line` is an empty list, `line[0]` would fail)